test/compilable/test18295.d

```diff
@@ -0,0 +1,19 @@
+// REQUIRED_ARGS: -dip1000
+
+// https://issues.dlang.org/show_bug.cgi?id=18295
+
+// See test/fail_compilation/test18295.d for the non-`-dip1000` version`
+
+scope class C { int i; }    // Notice the use of `scope` here
+
+C increment(return scope C c) @safe      // Should not emit error here
```

Comment:
Ideally we'd want to add this `-dip1000` test as well.

```d
scope class C { int i; }

C increment(return ref C c) @safe // check implicit scope from class type
{
    c.i++;
    return c;
}

void main() @safe
{
    scope C c = new C();
    c.increment();
}
```

Doesn't work though cause `scope class` isn't a real class reference.
```
bug.d(3): Error: variable bug.increment.c globals, statics, fields, manifest constants, ref and out parameters cannot be `scope`
```

Reply:
Then I suggest merging this and filing a bug.